node_macros/src/rust_type.rs

```diff
@@ -39,27 +49,45 @@ impl RustType {
             .clone()
     }
 
-    /// Returns the type without the path, but with type arguments, e.g. `SmallNet<NodeId>`.
-    pub fn ty(&self) -> TokenStream {
-        let ident = self.ident();
-        let args = &self
+    /// Returns the types identifier with type arguments attached as strings, i.e.
+    /// `GossipIncomingDeploy`'.
+    pub(crate) fn stringified_ident(&self) -> Ident {
+        let base = self
             .0
             .segments
             .last()
             .expect("type has no last part?")
-            .arguments;
+            .ident
+            .clone();
+
+        // This is a slightly ugly hack to get a string out of the type.
+        let final_arg = self.final_type_arg_ident();
+        let args: String = quote!(#final_arg)
+            .to_string()
+            .chars()
+            .filter(|c| c.is_alphabetic())
```

Comment:
Should this say "alphanumeric", since type identifiers can contain numbers?

Reply:
I hope we're not using the "alphanumericability", I am fine with this error'ing. If it was the case though, we'd have to check the first letter separately, since idents cannot _start_ with numbers.

For simplicity, I just disallowed numbers here.